monte_carlo.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np
+from tqdm import tqdm
+from collections import defaultdict
+import warnings
+
+from cell import CELL
+
+
+class Policy:
+    def __init__(self, train_epsilon, test_epsilon):
+        self.policy = {}
+    
+        self.train_epsilon = train_epsilon
+        self.test_epsilon = test_epsilon
+        self.epsilon = train_epsilon
+
+    def __getitem__(self, state):
+        if state not in self.policy:
+            self.policy[state] = np.random.choice(MonteCarlo.get_legal_actions(state))
+        if np.random.random() < self.epsilon:
+            return np.random.choice(MonteCarlo.get_legal_actions(state))
+        return self.policy[state]
+    
+    def __setitem__(self, state, action):
+        self.policy[state] = action
+
+    def __len__(self):
+        return len(self.policy)
+    
+    def set_trained(self):
+        self.epsilon = self.test_epsilon
+
+    def load(self, filename):
+        with open(filename, 'r') as f:
+            for line in f:
+                state = tuple(map(int, line.split(',')[:4]))
+                action = line.split(',')[4].strip()
+                self.policy[state] = action
+
+    def save(self, filename):
+        with open(filename, 'w') as f:
+            for state, action in self.policy.items():
+                f.write(f"{state[0]},{state[1]},{state[2]},{state[3]},{action}\n")
+    
+class MonteCarlo:
+
+    actions = {'increase_vx': (1, 0), 'decrease_vx': (-1, 0),
```

Comment:
can we simplify this with an simple array of  [[0,0], [0, 1], ...] and to only iterate over it when used in bulk